src/github.com/getlantern/flashlight/main/settings.go

```diff
@@ -2,198 +2,281 @@ package main
 
 import (
 	"io/ioutil"
-	"net/http"
 	"path/filepath"
+	"strings"
 	"sync"
 
 	"github.com/getlantern/appdir"
-	"github.com/getlantern/launcher"
 	"github.com/getlantern/yaml"
 
 	"github.com/getlantern/flashlight/ui"
 )
 
+// Notifier is what caller register to get notified when a settings entry changed
+type Notifier func(prev, cur interface{})
+
+// BoolNotifier is a shortcut of Notifier on bool entries
+type BoolNotifier func(cur bool)
+
 const (
+	ProxyAll    id = "proxyAll"
```

Comment:
Use predefined strings as key, but keep the type private so other packages can create keys on the fly. Not useful ATM.